libmamba/src/api/create.cpp

```diff
@@ -47,6 +47,15 @@ namespace mamba
                     LOG_ERROR << "Overwriting root prefix is not permitted";
                     throw std::runtime_error("Aborting.");
                 }
+                else if (!fs::is_directory(ctx.prefix_params.target_prefix))
+                {
+                    LOG_ERROR << "Target prefix already exists and is not a folder";
+                    throw std::runtime_error("Aborting.");
```

Comment:
Prefer `mamba::mamba_error` for exceptions thrown from `libmamba`. Also the exception should contain the same message as the log, so something like `"Target prefix already exists and is not a folder - aborting"`). That way the exception handling can use the message and debugging is easier.

Reply:
@Klaim Thanks for the review! I copied the pattern from lines 48 and 71 in this file, would you like me to update those as well?

Also, is `mamba_error_code::incorrect_usage` the proper error code to pass to `mamba_error()` in these cases?

Reply:
Yes please, not all the code of libmamba has been updated to this way indeed, and if it doesnt break tests I welcome applying the boyscout rule. Thanks! 👍🏼

I think `mamba_error_code::incorrect_usage` is correct here yes.

Reply:
@Klaim I think I've implemented what you suggested.